app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TweetController < ApplicationController
+  get '/tweets' do
+    if logged_in?
+      @all_tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+  
+  get '/tweets/new' do
+    if logged_in?
+      current_user
+      erb :'tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+  
+  post '/tweets' do
+    if params[:content] != "" 
+      Tweet.create(content: params[:content], user_id: current_user.id)
+    else
+      redirect '/tweets/new'
+    end
+  end
+  
+  get '/tweets/:tweet_id' do
+    if logged_in?
+      @tweet = Tweet.find_by(id: params[:tweet_id])
+    else
+      redirect '/login'
+    end
+    erb :'tweets/show_tweet'
+  end
+  
+  get '/tweets/:tweet_id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:tweet_id])
+    else
+      redirect '/login'
+    end
+    erb :'/tweets/edit_tweet'
+  end
+  
+  post '/tweets/:tweet_id' do
+    if logged_in?
+      redirect "tweets/#{params[:tweet_id]}/edit" if params[:content] == ""
+      @tweet = Tweet.find_by_id(params[:tweet_id])
+      @tweet.content = params[:content]
+      @tweet.save
+    else
+      redirect '/login'
+    end
+    erb :'tweets/show_tweet'
+  end
+  
+  post '/tweets/:tweet_id/delete' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:tweet_id])
+      @tweet.delete if @current_user.id == @tweet.user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.